test/optimisation/Optimisation.jl

```diff
@@ -0,0 +1,557 @@
+            loglikelihood_weight::T2
+        end
+        DynamicPPL.NodeTrait(::OverrideContext) = DynamicPPL.IsParent()
+        DynamicPPL.childcontext(parent::OverrideContext) = parent.context
+        DynamicPPL.setchildcontext(parent::OverrideContext, child) = OverrideContext(
+            child,
+            parent.logprior_weight,
+            parent.loglikelihood_weight
+        )
+
+        # Only implement what we need for the models above.
+        function DynamicPPL.tilde_assume(context::OverrideContext, right, vn, vi)
+            value, logp, vi = DynamicPPL.tilde_assume(context.context, right, vn, vi)
+            return value, context.logprior_weight, vi
+        end
+        function DynamicPPL.tilde_observe(context::OverrideContext, right, left, vi)
+            logp, vi = DynamicPPL.tilde_observe(context.context, right, left, vi)
+            return context.loglikelihood_weight, vi
+        end
+
+        @model function model1(x)
+            μ ~ Uniform(0, 2)
+            x ~ LogNormal(μ, 1)
+        end
+
+        @model function model2()
+            μ ~ Uniform(0, 2)
+            x ~ LogNormal(μ, 1)
+        end
+
+        x = 1.0
+        w = [1.0]
+
+        @testset "With ConditionContext" begin
+            m1 = model1(x)
+            m2 = model2() | (x=x,)
+            ctx = Turing.Optimisation.OptimizationContext(DynamicPPL.LikelihoodContext())
+            @test Turing.Optimisation.OptimLogDensity(m1, ctx)(w) ==
+                  Turing.Optimisation.OptimLogDensity(m2, ctx)(w)
+        end
+
+        @testset "With prefixes" begin
+            function prefix_μ(model)
+                return DynamicPPL.contextualize(
+                    model, DynamicPPL.PrefixContext{:inner}(model.context)
+                )
+            end
+            m1 = prefix_μ(model1(x))
+            m2 = prefix_μ(model2() | (var"inner.x"=x,))
+            ctx = Turing.Optimisation.OptimizationContext(DynamicPPL.LikelihoodContext())
+            @test Turing.Optimisation.OptimLogDensity(m1, ctx)(w) ==
+                  Turing.Optimisation.OptimLogDensity(m2, ctx)(w)
+        end
+
+        @testset "Weighted" begin
+            function override(model)
+                return DynamicPPL.contextualize(
+                    model,
+                    OverrideContext(model.context, 100, 1)
+                )
+            end
+            m1 = override(model1(x))
+            m2 = override(model2() | (x=x,))
+            ctx = Turing.Optimisation.OptimizationContext(DynamicPPL.DefaultContext())
+            @test Turing.Optimisation.OptimLogDensity(m1, ctx)(w) ==
+                  Turing.Optimisation.OptimLogDensity(m2, ctx)(w)
+        end
+    end
+
+    @numerical_testset "gdemo" begin
+        @testset "MLE" begin
+            Random.seed!(222)
+            true_value = [0.0625, 1.75]
+            true_logp = loglikelihood(gdemo_default, (s=true_value[1], m=true_value[2]))
+
+            function check_success(result)
```

Comment:
Are just redefining this method within every testset? If so, maybe we should define it once at the top of the file, and just have it take the `true_vals ` as an additional input

Reply:
Yeah, no idea why I was doing that. Fixed now.

Reply:
Fun side note, found this while fixing this: https://github.com/JuliaLang/julia/issues/54536